### ISSUE TITLE
Feature Request: Site admins to see all boards and change any board permissions

### ISSUE THREAD

**Issue by JackNWeems** [wekan/wekan#2413]
Site admins can't change board permissions unless the are an admin for that board. I deleted the user account for someone who was the only admin on one of our boards and as a result no one was able to change the settings for that board or add new users.  I think site admins should be able to see all boards that exist and change their settings.

**Comment by GCHPX** [comment]
I also have the same issue as @JackNWeems :/
is there a way to change the board admin to a different user manualy? I have wekan installed via snap.
Cheers.

**Comment by xet7** [comment]
@GCHPX 

Yes, do web search for nosqlbooster MongoDB GUI, and with it login via SSH to server localhost 27019 port where Wekan MongoDB is at database named Wekan, and edit boards tables where some other user is BoardAdmin. See also Users tables what users have what user ID.

There is no web UI for that yet.

**Comment by GCHPX** [comment]
Thank you so much @xet7 , I will try this right away!

**Comment by jurjendevries** [comment]
Hi all, I have a similar issue. I try to archive boards that have been created due subtasks. Unfortunately nosqlbooster isn't running on my Chromebook. I tried MongoDB Compass instead, but it didn't connect with my SSH tunnel. I don't know why because I don't get an error message.
So I tried to do this by the CLI with mongo shell. The shell started well, and I am able to use wekan db and execute commands such as db.users.find()

After creating an extra backup I tried the following command to update isAdmin: true for my userid at the boards where I am already a user:

 db.boards.updateMany(   { userid: { "xxx" } },   {     $set: { isAdmin: true }   } )
Where xxx is my user id I found with db.users.find()

I got this error:
2022-03-08T06:10:20.556+0100 E QUERY    [thread1] SyntaxError: missing : after property id @(shell):1:56

I did some google search to this error, but I am not experienced with MongoDB to understand it.
Anybody knows the correct mongo shell command for this? Thanks in advance.

**Comment by xet7** [comment]
@jurjendevries 

Does this help?

https://github.com/wekan/wekan/wiki/Forgot-Password

**Comment by xet7** [comment]
@jurjendevries 

Do you use correct database, like wekan? See at that Forgot Password page about `show dbs` to show databases, and `use wekan` to use wekan database. Similar like MySQL `CREATE DATABASE` command, MongoDB server can have many differently named databases.

**Comment by jurjendevries** [comment]
> @jurjendevries
> 
> Does this help?
> 
> https://github.com/wekan/wekan/wiki/Forgot-Password

@xet7 Thank you. Sorry for my late reply. I am using the correct 'wekan' database from the mongo shell instead of Robo 3T. No issues with password, just for having admin access for boards I didn't create myself. Normally I would ask the board owner, but in this case it are tens of boards automatically created due a bug with subtasks.

The https://github.com/wekan/wekan/wiki/Forgot-Password page helps partly. I scrolled to the bottom section "Don't have Admin Rights to board" and the steps went well, except the update query example for 3th step "And set yourself as admin." is missing in this documentation. I am to new to mongo to build this query myself an are avroid making a mistake, so it would help if this documentation have an update or updatemany query example where I can use my userId that I got from step 1.

**Comment by jurjendevries** [comment]
@xet7 I hope you are doing well. Today thanks to a friend I learned about the query. Unfortunatily I don't see an option to update the wiki with this, so I guess its limited to some admins/editors to do so? Would you like to add the query as step 3 "Set yourself as admin."
The query is:
```
db.boards.updateMany(
{ members: { $elemMatch: { userId: “USER-ID-HERE”, isAdmin: false } } },
{
$set: { “members.$.isAdmin”: true },
}
);
```

Thank you.

**Comment by xet7** [comment]
@jurjendevries 

Added to https://github.com/wekan/wekan/wiki/Forgot-Password#set-user-as-boardadmin-on-all-boards-user-is-member-of

**Comment by jurjendevries** [comment]
@xet7 Thanks, but now step 1 and 2, are not in the same location of the documentation as step 3, so it's confusing because all 3 steps are necessary.

**Comment by xet7** [comment]
@jurjendevries 

Please clone wiki:
````
git clone https://github.com/wekan/wekan.wiki
```
Edit that Forgot-Password.md file so that it would be less confusing, and add it in .zip attachment to comment to this issue. Thanks!

**Comment by xet7** [comment]
@ShiftcodeDream 

By default, user only sees boards user have created, or has been invited to.

If admin needs to see boards of user, at Admin Panel / People / People admin can search for that user, and at right dropdown menu click Impersonate User. Because it is security sensitive operation, that admin user ID and actions like export board etc is logged into database. There will be more logging options later.

**Comment by ShiftcodeDream** [comment]
This is not the question. We need a board admin interface to :
- Attribute a board to a newly created user (if users don't, admins should !)
- Find orphan boards and take control on them. You know, sometimes people quits their jobs, and nobody else can access the board administration.
- Archive / delete old unused board : sometimes users create a board in order to manage a project or an event. We can't trust users to clean up their old unused boards, but admin is responsible in case of lack of disk space !
- Have a look on how many boards we host, how many disk space it uses (namely for file attachments), last activity date, ...

In one word, manage and pilot the application.

To bypass our issue, can you please give us a script or mongodb request to add a user (given the user id) as an administrator of all boards, so we can call it weekly using crontab?
Thank you for your help.
